cabby/rvs/templates.py

```diff
@@ -0,0 +1,164 @@
+# coding=utf-8
+# Copyright 2020 Google LLC
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''Define templates for RVS.'''
+
+from nltk import CFG, Production
+from nltk.parse.generate import generate, Nonterminal
+import pandas as pd
+from typing import Dict, Sequence, Text
+
+from cabby.geo import geo_item
+
+
+# Terminals in the grammar.
+MAIN_NO_V = [
+    "CARDINAL_DIRECTION from MAIN_PIVOT for NUMBER_INTERSECTIONS intersections",
+    "CARDINAL_DIRECTION from MAIN_PIVOT for NUMBER_BLOCKS blocks",
+    "NUMBER_BLOCKS blocks past MAIN_PIVOT",
+    "NUMBER_INTERSECTIONS intersections past MAIN_PIVOT",
+    "past MAIN_PIVOT",
+    "CARDINAL_DIRECTION and past MAIN_PIVOT",
+    "past MAIN_PIVOT and continue to the next intersection",
+    "past MAIN_PIVOT and continue to the next block",
+]
+MAIN = [
+    ". When you pass MAIN_PIVOT, you'll be just \
+    NUMBER_INTERSECTIONS intersections away",
+    ". When you pass MAIN_PIVOT, you'll be just NUMBER_BLOCKS blocks away",
+]
+V1 = ['Go', 'Walk', 'Head']
+V2 = ['Meet at GOAL_LOCATION.', 'Come to GOAL_LOCATION.']
+NEAR_GOAL_END = [". GOAL_LOCATION will be near NEAR_PIVOT."]
+NEAR_GOAL_START = [". It will be near NEAR_PIVOT."]
+AVOID = [
+        '. If you reach BEYOND_PIVOT, you have gone too far.',
+         '']
+GOAL_END = ['and meet at GOAL_LOCATION.']
+GOAL = ["GOAL_LOCATION"]
+
+
+def add_rules(nonterminal_name: Text, 
+list_terminals: Sequence[Text]) -> Sequence[Production]:
+    """Create the production rules for a givn nonterminal and a
+     list of terminals corresponding to it.
+
+    Args:
+      nonterminal_name: The name of the nonterminal.
+      list_terminals: The list of terminals that for each one a rule with 
+      the nonterminal will be produced.
+    Returns:
+      A sequence of productions rules.
+
+    """
+    prods = []
+    for phrase in list_terminals:
+        rule = Production(Nonterminal(nonterminal_name), (phrase,))
+        prods.append(rule)
+    return prods
+
+
+def create_templates():
+  """Creates the templates from the grammar."""
+
+  prods = [
```

Comment:
Can you provide one liner for each of these that describes what they are capturing?

Reply:
added